bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostAsterism.scala

```diff
@@ -0,0 +1,368 @@
+
+  def xBinning: XBinning
+  def yBinning: YBinning
+}
+
+/** GHOST asterism model.
+  */
+object GhostAsterism {
+
+  /** The GHOST guide fibers can be enabled or disabled for each science target.
+    * Typically they are enabled for bright targets (< B mag 18 for standard
+    * resolution, < B mag 17 for high resolution) but disabled for faint targets.
+    * OIWFS state can also be explicitly disabled, say, in a crowded field where
+    * OIWFS is less effective.
+    */
+  sealed trait OiwfsState extends Product with Serializable
+
+  object OiwfsState {
+    case object Enabled  extends OiwfsState
+    case object Disabled extends OiwfsState
+
+    val enabled: OiwfsState  = Enabled
+    val disabled: OiwfsState = Disabled
+
+    val All = NonEmptyList(enabled, disabled)
+
+    implicit val EqualOiwfsState = Equal.equalA[OiwfsState]
+  }
+
+
+  /** GHOST targets are associated with a guiding state (enabled or disabled),
+    * referring to whether the dedicated guide fibers surrounding the science
+    * target should be used.
+    *
+    * There is a default guiding state based on magnitude, but this can be
+    * explicitly overridden.
+    */
+  final case class GhostTarget(target: SPTarget, explicitOiwfsState: Option[OiwfsState]) {
+
+    def coordinates(when: Option[Instant]): Option[Coordinates] =
+      target.getCoordinates(when.map(_.toEpochMilli))
+  }
+
+  object GhostTarget {
+
+    /** The magnitude at which OIWFS state is disabled by default at standard
+      * resolution.
+      */
+    val StandardResCutoff: Magnitude =
+      Magnitude(18.0, MagnitudeBand.B, None, MagnitudeSystem.Vega)
+
+    /** The magnitude at which OIWFS state is disabled by default at high
+      * resolution.
+      */
+    val HighResCutoff: Magnitude =
+      Magnitude(17.0, MagnitudeBand.B, None, MagnitudeSystem.Vega)
+
+    private def defaultOiwfsState(t: GhostTarget, cutoff: Magnitude): OiwfsState =
+      t.target.getMagnitude(MagnitudeBand.B).forall(_.value < cutoff.value) ? OiwfsState.enabled | OiwfsState.disabled
+
+    private def oiwfsState(t: GhostTarget, cutoff: Magnitude): OiwfsState =
+      t.explicitOiwfsState | defaultOiwfsState(t, cutoff)
+
+    /** Computes the OiwfsState for the given target in standard resolution mode. */
+    def standardResOiwfsState(t: GhostTarget): OiwfsState =
+      oiwfsState(t, StandardResCutoff)
+
+    /** Computes the OiwfsState for the given target in high resolution mode. */
+    def highResOiwfsState(t: GhostTarget): OiwfsState =
+      oiwfsState(t, HighResCutoff)
+  }
+
+
+  /** X-binning options. */
+  sealed abstract class XBinning(val intValue: Int) extends Product with Serializable {
+    override def toString: String =
+      intValue.toString
+  }
+
+  object XBinning {
+    case object One extends XBinning(1)
+    case object Two extends XBinning(2)
+
+    val one: XBinning = One
+    val two: XBinning = Two
+
+    val All = NonEmptyList(one, two)
```

Comment:
Yeah, here it could just as easily be `NonEmptyList(One, Two)` but in code like

````scala
  val binOpt: Option[XBinning] = ...
  binOpt.fold(XBinning.One) { b => ??? }
````

You have to explicitly tag `XBinning.One` as `XBinning` because it cannot infer that.  If you use `XBinning.one` though it picks up the right type.